src/whitenoise/app_update.rs

```diff
@@ -0,0 +1,531 @@
+        let v2 = Version::from_str("1.2.3").unwrap();
+
+        assert_eq!(v1, v2);
+        assert_eq!(v1.cmp(&v2), std::cmp::Ordering::Equal);
+        assert_eq!(v1.partial_cmp(&v2), Some(std::cmp::Ordering::Equal));
+    }
+
+    #[test]
+    fn test_version_compare_major_greater() {
+        // Test comparison when major version differs (lower major but higher minor/patch)
+        let v1 = Version::from_str("2.0.0").unwrap();
+        let v2 = Version::from_str("1.9.9").unwrap();
+
+        assert!(v1 > v2);
+        assert!(v2 < v1);
+        assert_eq!(v1.cmp(&v2), std::cmp::Ordering::Greater);
+    }
+
+    #[test]
+    fn test_version_compare_minor_greater() {
+        // Test comparison when major is equal but minor differs (same major, higher minor)
+        let v1 = Version::from_str("1.3.0").unwrap();
+        let v2 = Version::from_str("1.2.9").unwrap();
+
+        assert!(v1 > v2);
+        assert!(v2 < v1);
+    }
+
+    #[test]
+    fn test_version_compare_patch_greater() {
+        // Test comparison when major and minor are equal but patch differs (same major.minor, higher patch)
+        let v1 = Version::from_str("1.2.4").unwrap();
+        let v2 = Version::from_str("1.2.3").unwrap();
+
+        assert!(v1 > v2);
+        assert!(v2 < v1);
+    }
+
+    #[test]
+    fn test_version_compare_ordering_chain() {
+        // Test a chain of versions to verify correct ordering
+        let v0 = Version::from_str("0.0.1").unwrap();
+        let v1 = Version::from_str("0.1.0").unwrap();
+        let v2 = Version::from_str("1.0.0").unwrap();
+        let v3 = Version::from_str("1.0.1").unwrap();
+        let v4 = Version::from_str("1.1.0").unwrap();
+        let v5 = Version::from_str("2.0.0").unwrap();
+
+        assert!(v0 < v1);
+        assert!(v1 < v2);
+        assert!(v2 < v3);
+        assert!(v3 < v4);
+        assert!(v4 < v5);
+    }
+
+    /// Version Display Tests (Display implementation)
+
+    #[test]
+    fn test_version_display() {
+        // Test that Display formats the version correctly as "x.y.z"
+        let version = Version::from_str("1.2.3").unwrap();
+
+        assert_eq!(version.to_string(), "1.2.3");
+    }
+
+    #[test]
+    fn test_version_display_zeros() {
+        // Test Display with zero values
+        let version = Version::from_str("0.0.0").unwrap();
+
+        assert_eq!(version.to_string(), "0.0.0");
+    }
+
+    #[test]
+    fn test_version_display_large_numbers() {
+        // Test Display with large numbers
+        let version = Version::from_str("100.200.300").unwrap();
+
+        assert_eq!(version.to_string(), "100.200.300");
+    }
+
+    // Version Clone and Copy Tests (derived traits)
+
+    #[test]
+    fn test_version_clone() {
+        // Test that Clone works correctly (using explicit clone for trait verification)
+        let v1 = Version::from_str("1.2.3").unwrap();
+        #[allow(clippy::clone_on_copy)]
+        let v2 = v1.clone();
+
+        assert_eq!(v1, v2);
+    }
```

Comment:
I think this test provides little value and should be removed.
`Copy` implies `Clone`; testing derived trait with `#[allow(clippy::clone_on_copy)]` is a code smell